tests/integration/common/data/interfaces/test_collections.py

```diff
@@ -378,6 +379,21 @@ def test_move_question_with_dependencies(db_session, factories):
     move_question_up(q2)
 
 
+def test_raise_if_question_has_any_dependencies(db_session, factories):
+    form = factories.form.create()
+    user = factories.user.create()
+    [q1, q2] = factories.question.create_batch(2, form=form)
+
+    add_question_condition(q2, user, GreaterThan(minimum_value=1000, question_id=q1.id))
+
+    assert raise_if_question_has_any_dependencies(q2) is None
+
+    with pytest.raises(DependencyOrderException) as e:
+        raise_if_question_has_any_dependencies(q1)
+    assert e.value.question == q2
+    assert e.value.depends_on_question == q1
```

Comment:
these might need some ty ignores for now sadly

Reply:
how do you say "it's not a BaseException" in Rust